src/erliam_srv.erl

```diff
@@ -51,28 +52,29 @@ init([]) ->
             {ok, #state{}}
     end.
 
+-spec handle_call(term(), _, state()) -> {reply, ok | {error, not_implemented}, state()}.
 handle_call(invalidate, _From, State) ->
     update_credentials(),
     {reply, ok, State};
 handle_call(_Request, _From, State) ->
     {reply, {error, not_implemented}, State}.
 
+-spec handle_cast(_, state()) -> {noreply, state()}.
 handle_cast(_Msg, State) ->
     {noreply, State}.
 
+-spec handle_info(refresh | _, state()) -> {noreply, state()}.
 handle_info(refresh, State) ->
     maybe_update_credentials(),
     {noreply, State};
 handle_info(_, State) ->
     {noreply, State}.
 
+-spec terminate(_, state()) -> ok.
 terminate(_Reason, _State) ->
     ets:delete(?TAB),
     ok.
 
-code_change(_OldVsn, State, _Extra) ->
```

Comment:
Why this was removed? From the docs:

> This callback is optional, so callback modules need not export it. If a release upgrade/downgrade with Change={advanced,Extra} specified in the appup file is made when code_change/3 isn't implemented the process will crash with an undef exit reason.

Now the behavior changed unless I'm missing something.

Reply:
We don't use app updates and we never will. Therefore, if this function is called, it should crash since that will be *very* unexpected.